src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ScheduleService.kt

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.activities
+
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.BadDataException
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseActivity
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseSchedule
+
+private const val HOURS_AND_MINUTES = "([01]?[0-9]|2[0-3]):([0-5][0-9])"
+
+class ScheduleService {
+
+  fun mapSchedules(request: CreateActivityRequest, courseActivity: CourseActivity): List<CourseSchedule> =
+    request.schedules.map {
+
+      validateRequest(it, courseActivity)
+
+      val (startHour, startMinute) = getHoursAndMinutes(it.startTime)
+      val (endHour, endMinute) = getHoursAndMinutes(it.endTime)
+
+      CourseSchedule(
+        courseActivity = courseActivity,
+        scheduleDate = it.date,
+        startTime = it.date.atTime(startHour, startMinute),
+        endTime = it.date.atTime(endHour, endMinute),
+        slotCategoryCode = "AM" // TODO SDI-610 What rules should we apply to default this value?
```

Comment:
Not directly relevant, Ive created a Slot Category enum, which would simplify validation